generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -86,6 +113,19 @@ private void waitForAppToFinish() {
         } finally {
             taskProducer.shutdownNow();
             checkpointer.shutdownNow();
+
+            boolean checkpointHasFinished = false;
+            int exitStatus = 0;
+            try {
+                checkpointHasFinished = checkpointer.awaitTermination(APPLICATION_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

Reply:
So maybe you could wrap the InterruptedException in another exception, catch it and exit with -1?.